ddl/db_test.go

```diff
@@ -3752,6 +3770,17 @@ func (s *testDBSuite5) TestAddIndexForGeneratedColumn(c *C) {
 	s.tk.MustQuery("select id1 from gcai_table use index(idx1)").Check(testkit.Rows("6"))
 	s.tk.MustExec("admin check table gcai_table")
 }
+func (s *testSerialDBSuite) TestAddIndexForGeneratedColumn(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	save := config.GetGlobalConfig().AlterPrimaryKey
+	config.GetGlobalConfig().AlterPrimaryKey = false
```

Comment:
https://github.com/pingcap/tidb/pull/17964 This PR provides a way to update the global config without data race. Perhaps we should comply with it.

```
config.UpdateGlobal(func(conf *config.Config) {
		conf.AlterPrimaryKey = true
	})
defer config.RestoreFunc()()
```

Reply:
Sure